### PR TITLE
Added functionality to scrape branches

### LINKED ISSUE #148: Scrape the list of branches in a Repository
**Is your feature request related to a problem? Please describe.**
No, it's not a problem, but while working on the repos, a user might need to scrape the branches present in the Repository. 

I am referring to existing branches like `main`, `master`, `dev`, `feature`, `etc..`

**Describe the solution you'd like**
I will be scraping the branches from using a function `.get_branch()`

### I would like to work on this issue under JWOC 2023. Please assign me. @nikhil25803

### PR DESCRIPTION
**Pull request by pooranjoyb**
## Methods Used :
`.get_branch()` 

## Description
The method returns the list of all the branches in the user's Repository.

## To what class it belongs?:
`Repository`

## Demonstration video

[demo10.webm](https://user-images.githubusercontent.com/90945182/222917662-d00bb6c0-b253-4940-a016-51410f313a40.webm)

### Would be grateful if you could add the label of JWOC '23 if this PR is merged. @nikhil25803 😄

### CODE REVIEW

File: src/scrape_up/github/respository.py [hunk 1]
```diff
@@ -403,4 +403,17 @@ def get_environment(self):
...
+        data = BeautifulSoup(data.text, "html.parser")
+        try:
+            branch = data.find_all(class_="branch-name css-truncate-target v-align-baseline width-fit mr-2 Details-content--shown")
+            allBranches = []
+            for branchNames in branch:
+                allBranches.append(branchNames.text.strip())
+            return allBranches
+        except:
+            message = "Failed to fetch branches"
+            return message
```
Comment (nikhil25803) [inline src/scrape_up/github/respository.py:419]: Please follow the return type structure of the project, check other methods for reference.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

File: src/scrape_up/github/respository.py (hunk 1 above)
Comment (pooranjoyb) [reply src/scrape_up/github/respository.py]: Please review. I've updated the return type structure. @nikhil25803

File: src/scrape_up/github/respository.py [hunk 2]
```diff
@@ -403,4 +403,20 @@ def get_environment(self):
...
+
+    def get_branch(self):
+        data = requests.get(f"https://github.com/{self.username}/{self.repository}/branches")
+        data = BeautifulSoup(data.text, "html.parser")
+        try:
+            branch = data.find_all(class_="branch-name css-truncate-target v-align-baseline width-fit mr-2 Details-content--shown")
+            allBranches = []
+            for branchNames in branch:
+                allBranches.append(branchNames.text.strip())
+            return allBranches
```
Comment (nikhil25803) [inline src/scrape_up/github/respository.py:416]: Also, change the return type in the `try` block.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

File: src/scrape_up/github/respository.py (hunk 2 above)
Comment (pooranjoyb) [reply src/scrape_up/github/respository.py:416]: > Also, change the return type in the `try` block.

Okk, sure.

Comment (pooranjoyb) [reply src/scrape_up/github/respository.py]: Done with the bug fixes. :)

**Review by nikhil25803 — APPROVED** [review]

**Pull request merged**